### PR TITLE
Feature/legion shader language and compiler

### PR DESCRIPTION
**Pull request by GlynLeine**

### CODE REVIEW

File: applications/sandbox/engine/shaderlib/stdio.shinc [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+{
+    return normalize((stdin(0, modelMatrix) * vec4(dir, 0.0)).xyz);
+}
+
+vec3 GetWorldPoint(vec3 p)
+{
+    return (stdin(0, modelMatrix) * vec4(p, 1.0)).xyz;
+}
+
+mat3 GetTBN(int index)
```
Comment (kukash) [inline applications/sandbox/engine/shaderlib/stdio.shinc]: what does TBN mean? more descriptive name pls

Comment (GlynLeine) [reply applications/sandbox/engine/shaderlib/stdio.shinc]: TangentBitangentNormal matrix, it's a very common term within pbr though

**Review by kukash — COMMENTED** [review]
Maybe rename TBN, seems to be the only acronym used,

File: applications/sandbox/engine/shaderlib/stdio.shinc (hunk 1 above)
Comment (kukash) [reply applications/sandbox/engine/shaderlib/stdio.shinc]: google could not tell me but fine if its common

**Review by kukash — APPROVED** [review]

File: legion/engine/rendering/components/camera.hpp [hunk 2]
```diff
@@ -1,10 +1,61 @@
...
+                            math::vec3 vdir;
+                            float meta2;
+                        };
+                        math::vec4 vdirmeta;
+                    };
+                    uint idx;
+                };
+            };
+        };
+
```
Comment (Algo-ryth-mix) [inline legion/engine/rendering/components/camera.hpp:70]: requires docs

File: legion/engine/rendering/data/framebuffer.hpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            glNamedFramebufferRenderbuffer(id, attachement, GL_RENDERBUFFER, renderbuffer.id);
+        }
+
+        void attach(const )
+
+        void release()
+        {
+            glBindFramebuffer(target, 0);
+        }
+    };
```
Comment (Algo-ryth-mix) [inline legion/engine/rendering/data/framebuffer.hpp]: as does this

File: legion/engine/rendering/data/material.hpp [hunk 4]
```diff
@@ -130,13 +192,41 @@ namespace legion::rendering
...
         return MaterialCache::m_materials[id].get_param<T>(name);
     }
 
+    template<typename T>
+    T material_handle::get_param(GLint location)
+    {
+        async::readonly_guard guard(MaterialCache::m_materialLock);
+        return MaterialCache::m_materials[id].get_param<T>(location);
+    }
+
```
Comment (Algo-ryth-mix) [inline legion/engine/rendering/data/material.hpp:273]: more docs please

**Review by Algo-ryth-mix — CHANGES_REQUESTED** [review]
overall too little comments in the public interfaces.

**Review by Algo-ryth-mix — APPROVED** [review]

**Pull request merged**